debugger.md

```diff
@@ -0,0 +1,160 @@
+# Debugger proposal
+
+## Background
+
+One of the greatest advantages of being a log based workflow engine is that Temporal workflows are replayable.
+
+A developer can download workflow history and run it through the SDK replayer to materialize the workflow program's
+state.
+
+Workflows, being "just code", can be debugged with tools the developer is familiar with.
+
+## Features
+
+### Overview
+
+Typically, when looking at workflow history, it's clear which workflow task a failure occurs in but it's not immediately
+obvious which line of code that correlates to.
+
+This proposal discusses how we can make the debugging experience better for developers.
+
+### Supported languages
+
+Initially we will support Core based languages (TypeScript and Python) but Go and Java could also be supported with
+additional effort.
+
+The main advantage of going for Core based languages is that those language SDKs run Rust Core in a separate thread
+pool, allowing interaction with the runtime while a workflow is paused by a debugger.
+
+Targeting Core based SDKs will allow us to reuse a lot of the code for Ruby, .NET and possibly other SDKs if we decide
+to rewrite them to be backed by Core.
+
+### Supported IDEs
+
+Initially we will support Microsoft's Visual Studio Code only while keeping in mind other IDEs.
+
+The rationale being that at least for JavaScript VSCode is by far the most popular IDE and we'd get the most value
+investing in providing good DX for that environment.
+
+### Components
+
+- VSCode plugin
+  - Main panel / Workflow history panel
+  - Debug adapter
+- Core replay debug server
+- Lang middleware
+
+#### Main panel
+
+Has input to enter workflow ID and optional run ID with submit button to start a debug session.
```

Comment:
> enter workflow ID and optional run ID

We also need a way to set where the actual workflow code even is so the replayer can even be setup. Just a workflow ID and run ID is not enough to start a replayer.

Reply:
Hmm.. could be in the input there or plugin config file (where you'd set your interceptors and data converters).